include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge.h

```diff
@@ -149,17 +149,18 @@ constexpr static bool __can_use_ternary_op_v = __can_use_ternary_op<_Rng1DataTyp
 // Do serial merge of the data from rng1 (starting from start1) and rng2 (starting from start2) and writing
 // to rng3 (starting from start3) in 'chunk' steps, but do not exceed the total size of the sequences (n1 and n2)
 template <typename _Rng1, typename _Rng2, typename _Rng3, typename _Index, typename _Compare>
-void
+std::pair<_Index, _Index>
 __serial_merge(const _Rng1& __rng1, const _Rng2& __rng2, _Rng3& __rng3, const _Index __start1, const _Index __start2,
-               const _Index __start3, const _Index __chunk, const _Index __n1, const _Index __n2, _Compare __comp)
+               const _Index __start3, const _Index __chunk, const _Index __n1, const _Index __n2, _Compare __comp,
+               const _Index __n3 = 0)
 {
     const _Index __rng1_size = std::min<_Index>(__n1 > __start1 ? __n1 - __start1 : _Index{0}, __chunk);
     const _Index __rng2_size = std::min<_Index>(__n2 > __start2 ? __n2 - __start2 : _Index{0}, __chunk);
     const _Index __rng3_size = std::min<_Index>(__rng1_size + __rng2_size, __chunk);
 
     const _Index __rng1_idx_end = __start1 + __rng1_size;
     const _Index __rng2_idx_end = __start2 + __rng2_size;
-    const _Index __rng3_idx_end = __start3 + __rng3_size;
+    const _Index __rng3_idx_end = __n3 > 0 ? std::min<_Index>(__n3, __start3 + __rng3_size) : __start3 + __rng3_size;
```

Comment:
Can the default value of `__n3` be `__n1 + __n2` instead of `0` to avoid this ternary operator?

Reply:
An additional suggestion would be to move the logic of calculating the output sequence size at `__rng3_size` calculation and leave that line intact.

Reply:
your concrete suggestion?

Reply:
I discussed this topic with Sergey, and he pointed out that `n3 = n1 + n2` as a default value will break the logic of the function. So have have nothing concrete to propose.

According to him  `n3` option is "an additional and optional limitation for the output range size, where 0 is a sign that the option should be disregarded".  Double purpose of it confuses me (0 as a size and a sign to ignore the option), but I have nothing better to suggest. It is not a blocker for the PR.